### DISCUSSION TITLE
Anytype app lock-In - please help me regain control of my data

### DISCUSSION THREAD

**Discussion by tempuserr** [anyproto/contributors#88] [Q&A]
1. Is it possible to export all notes from the Android app in any format?

I started using Anytype 0.25.5 on an internet-isolated smartphone, so cloud sync is not an option. I have created numerous notebooks and notes locally on my smartphone, but I can't find an option to export all of my data to another device. Additionally, I can't copy all note content to the clipboard; copying is only available on a per-line basis. If I have many paragraphs in a page or note, selecting them with the orange frame doesn't copy the contents. Copying all notes line by line is not a feasible option.

Or maybe I overlook the export option in the mobile app?

I also tried to synchronize the desktop client over the local network (both devices connected to the same router, also Internet-isolated) by entering the Recovery Phrase. However, the desktop client didn't allow any sync without connecting to the internet and did not synchronize with the smartphone. 
Could it be that claims of local network synchronization are not accurate? And an Anytype account and synchronization to the cloud seems to be necessary and forced to use local sync on more than one device?

2. Alternatively, is it possible to copy my Anytype profile and notes data from my smartphone's storage and open it directly on another device?
Are Android "raw" files the same as desktop ones?

I feel that my data is locked-in by this app. 
Is there a way to break free from this confinement and export all my my data without breaking Internet-isolation of devices?

**Comment by isle9** [comment]
1. Export is [not yet available](https://doc.anytype.io/anytype-docs/miscellaneous/feature-list-by-platform#mobile-features) on mobile apps.
2. Local sync should work iirc, and you could always just self host your backup node if you are having issues. For [more info](https://doc.anytype.io/anytype-docs/data-and-security/data-storage-and-deletion/self-hosting).

**Reply by fuksman** [reply]
Just to add about local sync: @tempuserr, that is either a bug or overly strict restrictions. Local sync between devices should work fine without external connections. From a technical perspective it relies on mDNS. Is there a chance that you have prohibited such connections at the network level?

**Comment by tempuserr** [comment]
Thanks for pointing out this features table.

ad. 2. I get error when I try to Login with Recovery Phrase in Windows client on same local network as Android client (checked on two different Windows 10 machines):
![anytype_windows_error](https://github.com/anyproto/contributors/assets/12864611/dea95b5a-ae9e-40fe-92da-a2784c13c051)
I didn't dig into this, but assume - maybe wrong - that _clientspace_ connection means that connection to backup/cloud node is needed.

I will check all network restrictions.
Since these are Internet-isolated devices anyway, I will try to disable all Windows firewall and will check local sync again.
Thanks for all tips.
I'll check it.

**Reply by tempuserr** [reply]
I checked my configuration:
- computers are discoverable on the LAN (the router doesn't have any settings turned on to block it).
- on Windows, LAN is set as a private trusted network.
- on the smartphone with Android 9, default network settings are used (no firewall, etc.).

I turned off the firewall on Windows and ran both the desktop and Android clients connected to the same router. On Android, the orange dot in the upper left corner displayed "Syncing..." before connecting to LAN (but after quitting app in apps manager and restarting it displayed "Preparing..." and hung on this message). 
Then I tried logging in with the same Recovery Phrase on the Windows client. The same error occurred.

Later, I'll attempt to reinstall the Windows client - maybe I messed something with installation.
edit: and later I'll try to run [Anysync Netcheck](https://community.anytype.io/t/anysync-netcheck-tool-instructions/10307) tool.

Anyway thanks for confirming that local sync should work even in Internet-isolated environment. I will investigate my network configuration deeply.

**Reply by isle9** [reply]
The 'unable to connect' issue might be related to this previous [forum topic](https://community.anytype.io/t/client-clientspace-unable-to-connect-fixed/10787). Here's a [list of things](https://doc.anytype.io/anytype-docs/miscellaneous/troubleshooting) you can do yourself to try and troubleshoot it. The last one seems to have fixed the issue for the person that posted the linked topic.

**Reply by tempuserr** [reply]
OK. I'll investigate further.

1. But I have question: assume that
- there is no Internet connection to any backup node
- Windows client have proper working connection to LAN network
- there are no other clients in the LAN
- user enter own, proper Recovery Phrase
how should the Anytype's startscreen look like after hitting _Login_ on Windows client? Empty homescreen with no notes? Or something else?
(now I have _Entering the Void..._ for few seconds and finally _client.clientspace: unable to connect_) What should I expect?


2. In meanwhile I run AnySync-NetCheck.

On isolated machine gives output like this:
![anytype_netcheck](https://github.com/anyproto/contributors/assets/12864611/11994fea-ab45-442a-b5dc-affe9a3c407a)

Is [this tool](/t/anysync-netcheck-tool-instructions/10307) still functional?
Because on other machine with full Internet access it still gives errors:
![anytype_full_access](https://github.com/anyproto/contributors/assets/12864611/1591fec1-703d-453d-bf98-783ec488041a)

3. How make account on community.anytype.io?

**Reply by isle9** [reply]
1. It should indeed say Entering the Void... and then log you into your account. It's just bugged for you unfortunately.
2. Seems like the link gives a 404 error, but I'm guessing you are linking to the Netcheck tool which should be functional.
3. You can find how to create an account [here](https://doc.anytype.io/anytype-docs/community/community-forum). Please create a bug report, and include your technical details + the screenshots from the Netcheck tool.
